site/app/controllers/student/SubmissionController.php

```diff
@@ -1020,6 +1020,80 @@ public function ajaxRegrade($gradeable_id): array {
         return $this->core->getOutput()->renderResultMessage($msg, true);
     }
 
+    /**
+     * @return JsonResponse
+     * @param string $gradeable_id
+     */
+    #[Route('/api/{_semester}/{_course}/gradeable/{gradeable_id}/values', methods: ['GET'])]
+    public function ajaxGetGradeableValues(string $gradeable_id): JsonResponse {
+        $user_id = $_GET['user_id'] ?? '';
+        // Faculty and Superusers can get values for other users, if it is a student, require the $_GET user id to be the same as the
+        // API authenticated user.
+        if ($this->core->getUser()->getAccessLevel() === User::LEVEL_USER && ($user_id !== $this->core->getUser()->getId())) {
+            return JsonResponse::getFailResponse('API key and specified user_id are not for the same user.');
+        }
```

Comment:
This permission should be based on the user's course level (instructor, full access grader, limited access grader, or user) rather than system level access.

Reply:
This has been changed to course level, I believe.